emacs: bump elisp packages

stupidcomputer opened the pull request:
Runs `pkgs/applications/editors/emacs/elisp-packages/update` to update Emacs package data. 

related: #466358 and #465282.
## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

stupidcomputer committed:
melpa-packages: updated 2025-11-29 (from overlay)

stupidcomputer committed:
elpa-packages: updated 2025-11-29 (from overlay)

stupidcomputer committed:
elpa-devel-packages: updated 2025-11-29 (from overlay)

stupidcomputer committed:
nongnu-packages: updated 2025-11-29 (from overlay)

stupidcomputer committed:
nongnu-devel-packages: updated 2025-11-29 (from overlay)

jian-lin commented:
Thanks for the PR.  I planned to do the bump immediately after [25.11 branch-off][0], but I do not have enough free time recently.

I have verified that this diff is the same as my local run of `update-from-overlay` against `emacs-overlay` `6726be2908c4c3461d08f7aadcacc17967198d28`.

As [usual], we should identify new failed builds after this bump and either fix them locally or report them to upstream.  I am building all elisp packages now to find new failed builds.

---

This PR probably contains breaking changes, so it is not allowed to be backported to stable release 25.05 to fix #466358 and #465282.  A PR bumping only `org` (and maybe also other packages needed by the bump of `org`) is suitable for #466358 and #465282.

[usual]: https://github.com/NixOS/nixpkgs/pull/450556
[0]: https://www.github.com/NixOS/nixpkgs/issues/443568

jian-lin committed:
emacsPackages.sdml-mode: remove a merged-by-upstream patch

jian-lin committed:
emacs: respect overrideAttrs when calculating melpaVersion

This fixes builds of: rtags, ac-rtags, company-rtags, flycheck-rtags,
helm-rtags, ivy-rtags and malinka.

jian-lin committed:
emacsPackages.gnosis: fix build

jian-lin committed:
emacsPackages.mu4e-walk: fix build

jian-lin committed:
emacsPackages.ws-butler: work around build failure[1]

[1]: https://github.com/NixOS/nixpkgs/issues/436534

jian-lin committed:
emacsPackages.nordic-night: 20251129.409 -> 20250624.1732

This downgrade fixes build. Downgrade because the author deleted some
commits and force-pushed[1].

[1]: https://codeberg.org/ashton314/nordic-night/issues/7

jian-lin committed:
emacsPackages.tramp: work around build failure

jian-lin commented:
New build failures of this bump:
- [x] emacs-ac-rtags-2.41-unstable-2025-11-19.drv
- [x] emacs-company-rtags-2.41-unstable-2025-11-19.drv
- [x] emacs-flycheck-rtags-2.41-unstable-2025-11-19.drv
- [x] emacs-gnosis-20251108.939.drv
- [x] emacs-helm-rtags-2.41-unstable-2025-11-19.drv
- [x] emacs-ivy-rtags-2.41-unstable-2025-11-19.drv
- [x] emacs-malinka-20171202.1021.drv
- [x] emacs-mu4e-walk-20251128.1525.drv
- [x] emacs-nordic-night-theme-20251129.409.drv: [reported](https://codeberg.org/ashton314/nordic-night/issues/7) to upstream
- [x] emacs-rtags-2.41-unstable-2025-11-19.drv
- [x] emacs-sdml-mode-20251030.1814.drv
- [x] emacs-tramp-2.8.0.4.drv
- [x] emacs-tramp-auto-auth-20191027.1419.drv
- [x] emacs-ws-butler-20250310.205.drv
- [x] emacs-yatex-20250224.1034.drv: cannot download src, leave it broken

I have fixed them all.
